Directory.Build.props

```diff
@@ -456,17 +455,13 @@
   </ItemDefinitionGroup>
 
   <!-- Import it at the end of the props file to override the OutputPath for reference assemblies and use common directory props -->
-  <Import Project="$(MSBuildThisFileDirectory)eng\ReferenceAssemblies.props" />
+  <Import Project="$(RepositoryEngineeringDir)ReferenceAssemblies.props" />
 
   <PropertyGroup Condition="'$(IsSourceProject)' == 'true'">
     <!-- Set the documentation output file globally. -->
     <DocumentationFile Condition="'$(DocumentationFile)' == ''">$(OutputPath)$(MSBuildProjectName).xml</DocumentationFile>
   </PropertyGroup>
 
-  <!-- Use Roslyn Compilers to build -->
-  <Import Project="$(RoslynPropsFile)" Condition="'$(RunningOnUnix)'!='true' and Exists('$(RoslynPropsFile)') and '$(UseRoslynCompilers)'!='false' and '$(RoslynIncompatibleMsbuildVersion)' != 'true'" />
```

Comment:
I never took time to understand why we need this here. Can you provide some context please? We have the same in dotnet/standard: https://github.com/dotnet/standard/blob/master/Directory.Build.props#L140

Reply:
It was first introduced here and never removed: https://github.com/dotnet/corefx/commit/67fd8265894bd6707a10517133001b143f747693.  At that point it looks like CoreFx itself was restoring the roslyn compilers and this was doing the work to hook it into the build.  Folks just kept updating that rather than fixing it by pushing the logic to buildtools.

It appears to have been broken for quite some time because we aren't restoring the package version it tries to use.  Instead we'll end up using the default version in the acquired SDK.  I examined some builds locally before my change and none are actually importing these.  @safern will lend a hand to check the official builds and confirm this is the case.

Even if we did want a specific version of the compilers this isn't the way to do it, we should specify [UsingToolMicrosoftNetCompilers](https://github.com/dotnet/arcade/blob/f7fe38eec19f51b76a7f7c1716ba8b982c7ac121/src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props#L44-L48)  and [MicrosoftNETCoreCompilersVersion](https://github.com/dotnet/arcade/blob/f7fe38eec19f51b76a7f7c1716ba8b982c7ac121/src/Microsoft.DotNet.Arcade.Sdk/tools/Compiler.props#L10).